xsdk: add  version @0.7.0

balay opened the pull request:
We hope to complete xsdk@0.7.0 in a week or so

Some of the changes were reviewed at https://github.com/spack/spack/pull/26725

cc: @ulrikeyang

Previously  this PR included the following changes - that are now split up (so they can be independently reviewed/merged)
- https://github.com/spack/spack/pull/27267
- https://github.com/spack/spack/pull/27303
- https://github.com/spack/spack/pull/27296
- https://github.com/spack/spack/pull/27295

balay committed:
xsdk: add version @0.7.0
- add xsdk_depends_on() that propagates cuda, rocm options
- enable +cuda for hypre, mfem, amrex
- enable +rocm for mfem, sundials, magma, amrex, tasmanian, ginkgo, heffte
- deprecate @0.5.0
- remove 'debug' variant as it doesn't really exist
- heffte: remove binding to openmpi+cuda
- dtk: use 3.1-rc2 on macos (3.1-rc3 elsewhere)
- petsc: disable trilinos

Co-authored-by: Cody Balos <balos1@llnl.gov>

jppelteret commented on var/spack/repos/builtin/packages/dealii/package.py:
```diff
@@ -25,6 +25,7 @@ class Dealii(CMakePackage, CudaPackage):
     generator = 'Ninja'
 
     version('master', branch='master')
+    version('9.3.2', branch='dealii-9.3')
```
We're currently preparing the next point release, so I'd prefer it if we did not target the git branch. We're currently running tests on the branch, which should be complete in the next day or two. I'll provide you a `sha` when it's done.

balay replied:
Sure - this is a temporary place-holder (for testing) until deal@9.3.2 is released.

Once released - it can be a separate PR to spack:develop. And once it gets merged in - this branch can be rebased (to remove this change)

jppelteret replied:
I've opened #27303 for this change.

balay commented:
I've removed the parts that are already in separate PRs,  collapsed the commits (updated the commit message) and removed the 'draft' status.

The CI is clean now - and this PR  ready for review/approval!

cc: @ulrikeyang
